server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStatusApisIT.java

```diff
@@ -361,6 +361,49 @@ public void testGetSnapshotsMultipleRepos() throws Exception {
         }
     }
 
+    public void testGetSnapshotsWithSnapshotInProgress() throws Exception {
+        createRepository("test-repo", "mock", Settings.builder().put("location", randomRepoPath()).put("block_on_data", true));
+
+        createIndex("test-idx-1");
+        ensureGreen();
+
+        logger.info("--> indexing some data");
+        for (int i = 0; i < 100; i++) {
+            indexDoc("test-idx-1", Integer.toString(i), "foo", "bar" + i);
+        }
+        refresh();
+
+        logger.info("--> snapshot");
+        ActionFuture<CreateSnapshotResponse> createSnapshotResponseActionFuture = startFullSnapshot("test-repo", "test-snap");
+
+        logger.info("--> wait for data nodes to get blocked");
+        waitForBlockOnAnyDataNode("test-repo");
+        awaitNumberOfSnapshotsInProgress(1);
+
+        GetSnapshotsResponse response1 = client().admin().cluster().prepareGetSnapshots("test-repo").setSnapshots("test-snap")
+            .setIgnoreUnavailable(true)
+            .execute().actionGet();
+        List<SnapshotInfo> snapshotInfoList = response1.getSnapshots("test-repo");
+        assertEquals(1, snapshotInfoList.size());
+        assertEquals(SnapshotState.IN_PROGRESS, snapshotInfoList.get(0).state());
+
+        String notExistedSnapshotName = "snapshot_not_exist";
+        GetSnapshotsResponse response2 = client().admin().cluster().prepareGetSnapshots("test-repo").setSnapshots(notExistedSnapshotName)
+            .setIgnoreUnavailable(true)
+            .execute().actionGet();
```

Comment:
Let's use `.get()` instead to not swallow stack traces (action get unwraps the future's `ExecutionException`).